titus-server-federation/src/main/java/com/netflix/titus/federation/service/FallbackJobServiceGateway.java

```diff
@@ -1,64 +1,272 @@
+        return getFallbackCompletable(methodName, primaryCompletable, secondaryCompletable);
+    }
+
+    @Override
+    public Completable deleteTaskAttributes(TaskAttributesDeleteRequest request, CallMetadata callMetadata) {
+        final String methodName = "deleteTaskAttributes";
+        Completable primaryCompletable = primary.deleteTaskAttributes(request, callMetadata);
+        Completable secondaryCompletable = secondary.deleteTaskAttributes(request, callMetadata);
+        return getFallbackCompletable(methodName, primaryCompletable, secondaryCompletable);
+    }
+
+    @Override
+    public Completable moveTask(TaskMoveRequest taskMoveRequest, CallMetadata callMetadata) {
+        final String methodName = "moveTask";
+        Completable primaryCompletable = primary.moveTask(taskMoveRequest, callMetadata);
+        Completable secondaryCompletable = secondary.moveTask(taskMoveRequest, callMetadata);
+        return getFallbackCompletable(methodName, primaryCompletable, secondaryCompletable);
+    }
+
+    private <T> Observable<T> getFallbackObservable(String methodName, Observable<T> primary, Observable<T> secondary) {
+        if (federationConfiguration.isRemoteFederationEnabled()) {
+            return primary.onErrorResumeNext( (t) -> {
+                incrementFallbackCounter(methodName, t);
+                if (shouldFallback(t)) {
+                    return secondary;
+                } else {
+                    return Observable.error(t);
+                }
+            });
+        } else {
+            return secondary;
+        }
+    }
+
+    private <T> Mono<T> getFallbackMono(String methodName, Mono<T> primary, Mono<T> secondary) {
+        if (federationConfiguration.isRemoteFederationEnabled()) {
+            return primary.onErrorResume(t -> {
+                incrementFallbackCounter(methodName, t);
+                if (shouldFallback(t)) {
+                    return secondary;
+                } else {
+                    return Mono.error(t);
+                }
+            });
+        } else {
+            return secondary;
+        }
+    }
+
+    private Completable getFallbackCompletable(String methodName, Completable primary, Completable secondary) {
+        return getFallbackObservable(methodName, primary.toObservable(), secondary.toObservable()).toCompletable();
+    }
+
+    private boolean shouldFallback(Throwable t) {
+        return t instanceof StatusRuntimeException && fallbackCodes.contains(Status.fromThrowable(t).getCode());
+    }
+
+    private void incrementFallbackCounter(String methodName, Throwable t) {
+        Id metricId = this.getMetricId(METRIC_FALLBACK_ROOT + methodName, t);
+        this.titusRuntime.getRegistry().counter(metricId).increment();
+    }
+
+    private Id getMetricId(String metricName, Throwable t) {
+        String reason = t.getClass().getSimpleName();
+
+        if (t instanceof StatusRuntimeException) {
```

Comment:
Remove this type check, `Status.fromThrowable()` already does it for you with the correct types. You might need to refactor this code a bit to recover the original message if `code == UNKNOWN`